balancer/rls/picker.go

```diff
@@ -65,8 +67,18 @@ type rlsPicker struct {
 	logger        *internalgrpclog.PrefixLogger
 }
 
+// isFullMethodNameValid return true if name is of the form `/service/method`.
+func isFullMethodNameValid(name string) bool {
+	// strings.Split("/service/method", "/") returns ["", "service", "method"].
+	return strings.HasPrefix(name, "/") && len(strings.Split(name, "/")) == 3
```

Comment:
`== 3` or `> 2`?  Does RLS's design not work with `/service/path/to/method`?

Also `strings.Count` would be slightly better than `Split` to avoid slice allocations that just get thrown away.

Reply:
Done for both. Oversight on my part to assume that `/service/path/to/method` is not a valid one.

Reply:
Proto never does this, so it's not something we usually consider.  Which is why I'm not sure if there are other assumptions baked in that `method` may not contain slashes, either in the basic design of RLS or your implementation here.

Reply:
There are no assumptions except when we lookup the incoming RPC path in the map of keybuilders, we do the same thing that we do when we lookup the method config in service config:
https://github.com/grpc/grpc-go/blob/master/balancer/rls/internal/keys/builder.go#L123

Reply:
Wouldn't we consider the method everything after `/service/`, though, whereas that considers it only the part after the final slash?

If the design doesn't accommodate/consider 3+ slashes in the method then we should probably just reject those paths when doing a pick.

Reply:
We consider everything up to the last slash as the service here as well: https://github.com/grpc/grpc-go/blob/master/clientconn.go#L930

Am I missing something?

Reply:
I'm not sure whether that's the right way to interpret the path or the other way around.

Do other languages consider this?  How do they interpret it?  There isn't much on this in our documentation that I can find besides:

> Path is case-sensitive. Some gRPC implementations may allow the Path format shown above to be overridden, but this functionality is strongly discouraged. gRPC does not go out of its way to break users that are using this kind of override, but we do not actively support it, and some functionality (e.g., service config support) will not work when the path is not of the form shown above.

IMO we should either reject 3+ slashes or reconcile with the other languages on what 3+ slashes means (does the service or the method get extended to include the extra slashes?).

Reply:
Updated based on our offline discussion to fail picks which have paths with more than 2 slashes. Thanks.